### PR TITLE
stats: fix resource leak

### PR DESCRIPTION
**Pull request by lamxTyler**

### CODE REVIEW

File: statistics/boostrap.go [hunk 1]
```diff
@@ -54,6 +54,7 @@ func (h *Handle) initStatsMeta(is infoschema.InfoSchema) (statsCache, error) {
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
+	defer terror.Call(rc[0].Close)
```
Comment (shenli) [inline statistics/boostrap.go]: We should move the defer to line 54.

Comment (lamxTyler) [reply statistics/boostrap.go]: But if the err is not nil, `rc` will be nil.

Comment (zimulala) [reply statistics/boostrap.go]: We can add a check as following:
```
if len(rc) > 0{ // defer ... }
```

Comment (shenli) [reply statistics/boostrap.go]: How about if rc != nil {rc[0].Close()}

**Comment by coocood** [comment]
LGTM

**Pull request merged**